File: Objective-C/Patterns/MVC/ColorsController.m
#import "MVC.h"
#import "ColorsModel.h"
#import "ColorsController.h"
#import "InvalidRgbFormatException.h"

@implementation ColorsController

- (void)addColorWithName:(NSString *)name rgbValue:(NSString *)rgbValue
{
    if (![rgbValue hasPrefix:@"#"]) {
        THROW([[InvalidRgbFormatException alloc] 
                initWithName:@"InvalidRgbFormatException"
                reason:@"rgbValue needs to start with '#'"
                userInfo:nil]);
    }
    [[[self view] model] addColorWithName:name rgbValue:rgbValue];
}

@end

File: Java/org/oobench/patterns/mvc/ColorsController.java
package org.oobench.patterns.mvc;

import org.oobench.patterns.mvc.common.*;

public class ColorsController extends Controller {

    public ColorsController(View theView) {
        super(theView);
    }

    public void handleEvent(ControllerEvent theEvent) {
    }

    public void addColorWithName(String name, String rgbValue) 
            throws InvalidRgbFormatException {
        if (!rgbValue.startsWith("#")) {
            throw new InvalidRgbFormatException(
                    "rgbValue needs to start with '#'");
        }
        ((ColorsModel)getView().getModel()).addColorWithName(name, rgbValue);
    }

    public void update() {
    }
}
